### PR TITLE
Added integration test

### PR DESCRIPTION
**Pull request by pkj-m**
1. Added integration test for testing full stack. 
2. Modified the matrix2graph function to accept sparse matrices with elements <: Number, not just integers
3. Added LinearAlgebra and DiffEqDiffTools as dependencies.

### CODE REVIEW

File: src/matrix2graph.jl [hunk 1]
```diff
@@ -10,7 +10,7 @@ sparse matrix, columns are represented with vertices
 and 2 vertices are connected with an edge only if
 the two columns are mutually orthogonal.
 """
-function matrix2graph(SparseMatrix::SparseMatrixCSC{Int64,Int64})
+function matrix2graph(SparseMatrix::SparseMatrixCSC{T,Int64}) where T<:Number
```
Comment (ChrisRackauckas) [inline src/matrix2graph.jl]: Instead of `Int64`, use `Int`.

File: Project.toml [hunk 2]
```diff
@@ -4,7 +4,9 @@ authors = ["Pankaj Mishra <pankajmishra1511@gmail.com>"]
 version = "0.1.0"
 
 [deps]
+DiffEqDiffTools = "01453d9d-ee7c-5054-8395-0335cb756afa"
```
Comment (ChrisRackauckas) [inline Project.toml]: This only needs to be a test dependency

Comment (pkj-m) [reply Project.toml]: Done. Moved to extras.

File: src/matrix2graph.jl (hunk 1 above)
Comment (pkj-m) [reply src/matrix2graph.jl]: Changed to Int.

**Pull request merged**